### PR TITLE
Enpass: Update to 6.2.0.539 and improvements

### LINKED ISSUE #2452: enpass 6.0.4 fails to install
It is not possible to install enpass.

I get error - `Can't shim 'Enpass.exe': File doesn't exist.`

commands uninstall, cache rm - are not able to fix problem

```

PS C:\Windows\system32> scoop uninstall enpass
Uninstalling 'enpass' (6.0.4.281).
Unlinking ~\scoop\apps\enpass\current
'enpass' was uninstalled.
PS C:\Windows\system32> scoop cache rm enpass
PS C:\Windows\system32> scoop install enpass
Installing 'enpass' (6.0.4.281) [64bit]
EnpassPackage-6.0.4.281.zip (54.0 MB) [=======================================================================] 100%
Checking hash of EnpassPackage-6.0.4.281.zip ... ok.
Extracting EnpassPackage-6.0.4.281.zip ... done.
Running installer script...
Linking ~\scoop\apps\enpass\current => ~\scoop\apps\enpass\6.0.4.281
Creating shim for 'Enpass'.
Can't shim 'Enpass.exe': File doesn't exist.
PS C:\Windows\system32> ls ~\scoop\apps\enpass\current

Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----        1/30/2019   7:52 AM           4928 0
-a----       11/18/2017   1:00 PM         179200 u0
-a----        7/11/2017   6:39 AM           8332 u1
-a----       11/18/2017  12:57 PM           4180 u2
-a----       10/12/2018   6:42 AM           6966 u3
-a----       10/12/2018   6:42 AM           5768 u4
-a----        1/30/2019   7:52 AM           3682 u5

```

### PR DESCRIPTION
**Pull request by The-Loeki**
Enpass 6 on scoop was always broken for me due to the new packaging method. This fixes it and updates Enpass to 6.1. 

Closes #2452

### CODE REVIEW

File: bucket/enpass.json [hunk 1]
```diff
@@ -1,13 +1,31 @@
...
     "installer": {
         "script": [
-            "Expand-7zipArchive \"$dir\\Enpass*setup.exe\" \"$dir\"",
-            "Remove-Item \"$dir\\Enpass*setup.exe\""
+            "Expand-DarkArchive \"$dir\\Enpass-setup.exe\" \"$dir\\tmp\"",
+            "Expand-MsiArchive \"$dir\\tmp\\AttachedContainer\\Enpass.msi\" \"$dir\\tmp\" | Out-Null",
+            "movedir \"$dir\\tmp\\PFiles\\Enpass\" \"$dir\" | Out-Null"
+        ]
+    },
+    "post_install": [
```
Comment (Ash258) [inline bucket/enpass.json]: I do not like forcing registry manipulation for users. 

That is the reason why I use scoop. To ditch installers bad behaviour of setting registry.

See vscode, tortoisesvn manifests how to implement this optionally.

Comment (The-Loeki) [reply bucket/enpass.json]: I can't say I disagree with a fair amount of conservatism on dumping stuff in the registry ;) 

In this case however I see no other way to fix this. 

Enpass has cloud sync capabilities as core functionality, for which it needs browser interaction to setup authentication.
AFAIK these handlers are the only way to get that working without things going confusingly wrong (they're the smallest subset of reg entries I could make).
Notice I also properly remove the keys as necessary, and they don't seem likely to interfere elsewhere.

So making it optional would only make installation more of a hassle and errorprone to the vast majority of the userbase.

Fortunately it's not like we're dumping 3 tons of uncleanable garbage in there for no apparent reason, unlike some apps we know.

Comment (The-Loeki) [reply bucket/enpass.json]: @Ash258 bump?

**Comment by The-Loeki** [comment]
Hi @Ash258; I think you notice I'm not an expert @ PowerShell there ;) registry entry LGTM though, it's identical to the code I submitted. 
I'll test it in practice and fix if it should be broken anyway, thanks!

**Comment by Ash258** [comment]
Add closing directive into PR descriptino for #2452

**Comment by The-Loeki** [comment]
Updated to 6.1.1.451, just succesfully installed over here.

**Comment by The-Loeki** [comment]
>     * Replace 2 spaces with 4. 
np, done (didn't see that in the checks, or was I not paying attention?)

> * Did you used checkver to update?
Negative, because I've had to change the URL to the setup.exe rather than the zip files, which seem gone.

> * Add closing directive for issue
Took me a while to figure what you meant there, but done ;)

File: bucket/enpass.json [hunk 2]
```diff
@@ -1,26 +1,48 @@
...
-    "bin": "Enpass.exe",
+    "license": {
+        "identifier": "Proprietary",
+        "url": "https://www.enpass.io/legal/"
+    },
     "installer": {
         "script": [
-            "Expand-7zipArchive \"$dir\\Enpass*setup.exe\" \"$dir\"",
-            "Remove-Item \"$dir\\Enpass*setup.exe\""
+            "Expand-DarkArchive \"$dir\\Enpass-setup.exe\" \"$dir\\tmp\" -Removal",
```
Comment (Ash258) [inline bucket/enpass.json]:
```suggestion
            "Expand-DarkArchive \"$dir\\$fname\" \"$dir\\tmp\" -Removal",
```

**Comment by Ash258** [comment]
https://github.com/lukesampson/scoop-extras/pull/2275#issuecomment-511139403

> Add closing directive into PR descriptino for #2452 

Add `- Closes #2452` to PR Description

When you change URL, first change autoupdate and then run `checkver` with force parameter

**Comment by The-Loeki** [comment]
> > Add closing directive into PR descriptino for #2452
> 
> Add `- Closes #2452` to PR Description
Ah, still misunderstood then; it's in the fd9b363 commit, I'll update the PR desc

> When you change URL, first change autoupdate and then run `checkver` with force parameter
```
PS>.\checkver.ps1 -App enpass -Dir H:\WS\_scratch\ -Force
enpass: 6.1.1.451 (scoop version is 6.1.1.451)
Forcing autoupdate!
Autoupdating enpass
Downloading Enpass-setup.exe to compute hashes!
Loading Enpass-setup.exe from cache
Computed hash: 132ba08f47bbf8799234f6931ab74bf58f15810110aeb117b67e82a2e5d0508b
Writing updated enpass manifest
```

**Comment by The-Loeki** [comment]
Updated Enpass to 6.2.0.539. Current repo still has the broken install. Is something still stopping this PR?

**Comment by Ash258** [comment]
I am still waiting for response on https://github.com/lukesampson/scoop-extras/pull/2275#pullrequestreview-259672881

Are you sure that URL Protocol key should be empty and default key should contain "URL:enapssauth Protocol"?

**Comment by The-Loeki** [comment]
Hi Ash,

Sorry about the misunderstanding then:
> 
> Updated to 6.1.1.451, just succesfully installed over here.

> Are you sure that URL Protocol key should be empty and default key should contain "URL:enapssauth Protocol"?

Yes, it's one of them Black Voodoo registry haxx ;)

File: bucket/enpass.json [hunk 3]
```diff
@@ -1,31 +1,48 @@
...
+    "url": "https://dl.enpass.io/stable/windows/setup/6.2.0.539/Enpass-setup.exe",
+    "hash": "4ad268b913f589513d167aa2591ed806cadcb0ec5e1c3154378ca334b5adc87b",
     "description": "A cross-platform password management app to securely store passwords and other credentials in a virtual vault locked with a master password.",
-    "version": "6.0.4.281",
+    "homepage": "https://www.enpass.io/",
     "license": {
-        "identifier": "Freeware",
+        "identifier": "Proprietary",
         "url": "https://www.enpass.io/legal/"
     },
```
Comment (Ash258) [inline bucket/enpass.json]:
```suggestion
    },
    "url": "https://dl.enpass.io/stable/windows/setup/6.2.0.539/Enpass-setup.exe",
    "hash": "4ad268b913f589513d167aa2591ed806cadcb0ec5e1c3154378ca334b5adc87b",
```

File: bucket/enpass.json [hunk 4]
```diff
@@ -1,31 +1,48 @@
 {
-    "homepage": "https://www.enpass.io/",
+    "version": "6.2.0.539",
+    "url": "https://dl.enpass.io/stable/windows/setup/6.2.0.539/Enpass-setup.exe",
```
Comment (Ash258) [inline bucket/enpass.json]:
```suggestion
```

File: bucket/enpass.json [hunk 5]
```diff
@@ -1,31 +1,48 @@
...
+    "hash": "4ad268b913f589513d167aa2591ed806cadcb0ec5e1c3154378ca334b5adc87b",
```
Comment (Ash258) [inline bucket/enpass.json]:
```suggestion
```

File: bucket/enpass.json [hunk 6]
```diff
@@ -1,31 +1,48 @@
...
     "description": "A cross-platform password management app to securely store passwords and other credentials in a virtual vault locked with a master password.",
-    "version": "6.0.4.281",
+    "homepage": "https://www.enpass.io/",
```
Comment (Ash258) [inline bucket/enpass.json]:
```suggestion
    "homepage": "https://www.enpass.io",
```

**Pull request merged**